### PR TITLE
Fix LR scheduler checkpoint resume

### PR DESCRIPTION
**Pull request by mikasenghaas**
This PR fixes setting up WSD LR scheduling across resuming borders. It makes sure that we can skip loading the LR scheduler from a checkpoint but setup a fresh instance, or load the exact same scheduler from the DCP checkpoint. 

In this [W&B project](https://wandb.ai/primeintellect/mika?nw=wqmcynf32x) is an example of a stage 1 with warmup+constant, a stage 2 with constant + decay and the stage 2 being resumed.

Screenshot 2025-10-13 at 11 42 25 AM

Screenshot 2025-10-13 at 11 42 33 AM

### CODE REVIEW

File: src/prime_rl/trainer/sft/train.py [hunk 1]
```diff
@@ -112,7 +118,18 @@ def train(config: SFTTrainerConfig):
...
+            [optimizer],
+            scheduler if not config.ckpt.skip_scheduler else None,
+            progress if not config.ckpt.skip_progress else None,
+            step=config.ckpt.resume_step,
+            dataloader=dataloader if not config.ckpt.skip_dataloader else None,
+        )
+        # This redundant setup is necessary because loading the optimizer's state has side effects on the scheduler state dict
+        if config.ckpt.skip_scheduler:
+            scheduler = setup_scheduler(optimizer, config.scheduler, scheduler_steps, config.optim.lr)
+        print(scheduler.state_dict())
```
Comment (samsja) [inline src/prime_rl/trainer/sft/train.py]:
```suggestion
```

**Review by samsja — APPROVED** [review]
lgtm
